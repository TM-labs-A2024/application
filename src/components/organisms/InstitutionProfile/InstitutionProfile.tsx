import { ArrowBackIcon } from '@chakra-ui/icons'
import { IconButton, Text, Stack, Heading, Divider } from '@chakra-ui/react'
import { Institution } from '@src/types'
import { isIOS, isAndroid } from '@utils/index'
import Image from 'next/image'
import { useRouter } from 'next/navigation'
import React from 'react'

export default function InstitutionProfile({ institution }: { institution: Institution }) {
  // --- Hooks -----------------------------------------------------------------
  const router = useRouter()
  // --- END: Hooks ------------------------------------------------------------

  return (
    <div
      className={`mx-auto block h-screen w-screen overflow-hidden px-8 ${isIOS() ? 'pb-64 pt-20' : 'pb-0 pt-10'} ${isAndroid() ?? 'pb-64 pt-8'}`}
      data-testid="institution-profile"
    >
      <div className={`mb-8 flex flex-row justify-between`}>
        <Image alt="logo" src="/static/images/logo-horizontal.png" width={200} height={80} />
      </div>
      <div className="mb-8 flex flex-row items-center justify-start">
        <IconButton
          size="xl"
          className="mr-4"
          aria-label="back"
          variant="link"
          icon={<ArrowBackIcon />}
          onClick={() => {
            router.back()
          }}
        />
        <Text className="font-medium">Perfil de la institución</Text>
      </div>
      <div className="flex h-4/5 flex-col justify-between lg:px-96">
        <div>
          <Stack spacing={1} mb={6}>
            <Heading as="h3" size="md" noOfLines={1}>
              {institution.name}
            </Heading>
            <Text>Credencial: {institution.credentials}</Text>
          </Stack>
          <Divider orientation="horizontal" />
          <Stack mb={6} mt={6}>
            <h4 className="text-sm text-gray-600">Dirección</h4>
            <Text className="font-medium">{institution.address}</Text>
          </Stack>
          <Stack mb={6} mt={6}>
            <h4 className="text-sm text-gray-600">Correo electrónico</h4>
            <Text className="font-medium">{institution.email}</Text>
          </Stack>
          <Stack mb={6}>
            <h4 className="text-sm text-gray-600">Teléfono</h4>
            <Text className="font-medium">{institution.phoneNumber}</Text>
          </Stack>
        </div>
      </div>
    </div>
  )
}
