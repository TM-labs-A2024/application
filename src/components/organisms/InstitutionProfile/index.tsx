import InstitutionProfile from './InstitutionProfile'

export default InstitutionProfile
