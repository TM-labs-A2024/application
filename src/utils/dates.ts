export const formatDate = (date: string) => {
  const chunks = date.split('-')

  const newDate = chunks[1] + '-' + chunks[2] + '-' + chunks[0]

  return newDate
}
