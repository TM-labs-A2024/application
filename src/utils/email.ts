import emailjs from '@emailjs/browser'

export function sendEmail(data: {
  from_name: string
  to_name: string
  code: string | number
  to_email: string
}) {
  const templateParams = {
    ...data
  }

  const serviceKey = process.env.NEXT_PUBLIC_EMAIL_SERVICE_KEY ?? ''
  const templateKey = process.env.NEXT_PUBLIC_EMAIL_TEMPLATE_KEY ?? ''
  const apiKey = process.env.NEXT_PUBLIC_EMAIL_API_KEY ?? ''

  emailjs.send(serviceKey, templateKey, templateParams, apiKey).catch(() => {
    alert('Algo sali√≥ mal, por favor reintenta el registro.')
  })
}
