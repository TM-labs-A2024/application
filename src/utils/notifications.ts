import { GENERIC_ERROR } from '@constants/index'
import { isMobile } from '@utils/index'
import { Store } from 'react-notifications-component'

export const setupErrorNotification = () => {
  Store.addNotification(
    GENERIC_ERROR('Algo salió mal, por favor inténtelo de nuevo.', isMobile(window))
  )
}
