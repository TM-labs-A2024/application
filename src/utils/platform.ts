import { Capacitor } from '@capacitor/core'

export const isWeb = () => Capacitor.getPlatform() === 'web'

export const isIOS = () => Capacitor.getPlatform() === 'ios'

export const isAndroid = () => Capacitor.getPlatform() === 'android'
