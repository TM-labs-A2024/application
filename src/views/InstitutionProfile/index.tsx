import InstitutionProfile from '@components/organisms/InstitutionProfile'
import { Institution } from '@src/types'
import React from 'react'

export default function InstitutionProfileView({
  institution,
  onLogout
}: {
  institution: Institution
  onLogout?: () => void
}) {
  return <InstitutionProfile institution={institution} onLogout={onLogout} />
}
