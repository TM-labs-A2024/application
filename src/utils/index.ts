import { formatDate } from './dates'
import { sendEmail, generateVerificationCode, setupEmailSending } from './email'
import { setupErrorNotification } from './notifications'
import { isWeb, isIOS, isAndroid, isMobile } from './platform'

export {
  isWeb,
  isIOS,
  isAndroid,
  isMobile,
  sendEmail,
  generateVerificationCode,
  setupEmailSending,
  setupErrorNotification,
  formatDate
}
