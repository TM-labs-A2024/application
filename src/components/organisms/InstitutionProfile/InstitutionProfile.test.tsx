import { institution } from '@constants/index'
import { render, screen } from '@test/utils/index'

import '@testing-library/jest-dom'
import InstitutionProfile from './InstitutionProfile'
import 'next/navigation'

jest.mock('next/navigation')

describe('Organisms > InstitutionProfile test', () => {
  test('The component shows the credentials, address, email and phone of the institution', () => {
    render(<InstitutionProfile institution={institution} />)

    const credentialsMatcher = (content: string) => content.includes(institution.credentials)
    const institutionCredentials = screen.getByText(credentialsMatcher)

    expect(institutionCredentials).toBeInTheDocument()
    expect(screen.getByText(institution.address)).toBeInTheDocument()
    expect(screen.getByText(institution.email)).toBeInTheDocument()
    expect(screen.getByText(institution.phone)).toBeInTheDocument()
  })

  test('Matches the snapshot', () => {
    render(<InstitutionProfile institution={institution} />)

    expect(screen.getByTestId('institution-profile')).toMatchSnapshot()
  })
})
