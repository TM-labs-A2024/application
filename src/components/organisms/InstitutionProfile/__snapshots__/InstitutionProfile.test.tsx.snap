// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organisms > InstitutionProfile test Matches the snapshot 1`] = `
<div
  class="mx-auto block h-screen w-screen overflow-hidden px-8 pb-0 pt-10 false"
  data-testid="institution-profile"
>
  <div
    class="mb-8 flex flex-row justify-between"
  >
    <img
      alt="logo"
      data-nimg="1"
      decoding="async"
      height="80"
      loading="lazy"
      src="/_next/image?url=%2Fstatic%2Fimages%2Flogo-horizontal.png&w=640&q=75"
      srcset="/_next/image?url=%2Fstatic%2Fimages%2Flogo-horizontal.png&w=256&q=75 1x, /_next/image?url=%2Fstatic%2Fimages%2Flogo-horizontal.png&w=640&q=75 2x"
      style="color: transparent;"
      width="200"
    />
  </div>
  <div
    class="mb-8 flex flex-row items-center justify-start"
  >
    <button
      aria-label="back"
      class="chakra-button mr-4 css-84vx7q"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="chakra-icon css-onkibi"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
          fill="currentColor"
        />
      </svg>
    </button>
    <p
      class="chakra-text font-medium css-0"
    >
      Perfil de la institución
    </p>
  </div>
  <div
    class="flex h-4/5 flex-col justify-between lg:px-96"
  >
    <div>
      <div
        class="chakra-stack css-1uriqkk"
      >
        <h3
          class="chakra-heading css-ypohwm"
        >
          Clínica Tex Mex
        </h3>
        <p
          class="chakra-text css-0"
        >
          Credencial: 
          14578245
        </p>
      </div>
      <hr
        aria-orientation="horizontal"
        class="chakra-divider css-svjswr"
      />
      <div
        class="chakra-stack css-1420a3n"
      >
        <h4
          class="text-sm text-gray-600"
        >
          Dirección
        </h4>
        <p
          class="chakra-text font-medium css-0"
        >
          Av. Fernandez Peña, 147.
        </p>
      </div>
      <div
        class="chakra-stack css-1420a3n"
      >
        <h4
          class="text-sm text-gray-600"
        >
          Correo electrónico
        </h4>
        <p
          class="chakra-text font-medium css-0"
        >
          test@test.com
        </p>
      </div>
      <div
        class="chakra-stack css-1j35777"
      >
        <h4
          class="text-sm text-gray-600"
        >
          Teléfono
        </h4>
        <p
          class="chakra-text font-medium css-0"
        >
          +582742214441
        </p>
      </div>
      <button
        class="chakra-button mt-56 w-full css-jzcxx5"
        type="button"
      >
        Cerrar sesión
      </button>
    </div>
  </div>
</div>
`
